Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +463,190 @@ def restore_snapshot_online(
         snapshot_name,
         {})
 
+def create_schedule(
+        client_obj,
+        storage_system_ip,        
+        storage_system_username,
+        storage_system_password,
+        schedule_name,
+        snapshot_name,
+        base_volume_name,
+        read_only,
+        expiration_time,
+        retention_time,
+        expiration_unit,
+        retention_unit,
+        task_freq, 
+        task_freq_custom):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule creation failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Schedule name is null",
+            {})                 
+    if snapshot_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Snapshot name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule creation failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+
+    if len(snapshot_name) < 1 or len(snapshot_name) > 20:
```

Comment:
Why are we checking against 20? Are we sure that the timestamp will always be 11 characters?

Reply:
we are considering maximum length of timestamp